internal/client/gitea.go

```diff
@@ -61,8 +62,50 @@ func (c *giteaClient) CreateFile(
 	path,
 	message string,
 ) error {
-	//TODO: implement for brew and scoop support for Gitea-hosted repos
-	return nil
+	// Gitea >= 1.12.0 Use DefaultBranch on ""
+	branchName := ""
+	if err := c.client.CheckServerVersionConstraint(">=1.12.0"); err != nil {
```

Comment:
Is this because gitea versions older than 1.12 didn't have the concept of default branches at all, or was it just not exposed in the API? What is the meaning of the empty string in these older versions?

Either way I assume it's correct, but it would be great to explain the motivation behind this a bit more in the comment.

Reply:
this wont be needed for v0.12.1
-> https://gitea.com/gitea/go-sdk/pulls/352

I'll add a ToDo comment to remove this when upgrading to gitea sdk v0.12.1